src/languageservice/services/yamlSchemaService.ts

```diff
@@ -282,6 +293,30 @@ export class YAMLSchemaService extends JSONSchemaService {
             return resolveSchema();
         }
     }
+    
+    /**
+     * Retrieve schema if declared as modeline.
+     * Public for testing purpose, not part of the API.
+     * @param doc 
+     */
+    public getSchemaFromModeline(doc: any) : string{
+        if (doc instanceof SingleYAMLDocument) {
+            const yamlLanguageServerModeline = doc.lineComments.find(lineComment => {
+                const matchModeline = lineComment.match(/^#\s+yaml-language-server\s*:/g);
+                return matchModeline !== null && matchModeline.length === 1;
+            });
+            if (yamlLanguageServerModeline != undefined) {
+                const schemaMatchs = yamlLanguageServerModeline.match(/\$schema=\S+/g);
+                if(schemaMatchs !== null && schemaMatchs.length >= 1) {
+                    if (schemaMatchs.length >= 2) {
+                        console.log('Several $schema attributes has been found on the yaml-language-server modeline. The first one will be picked.');
```

Comment:
How can I send a notification to the client?

Reply:
This can be fine for now. server.ts binds `console.log` and `console.error` to the connection.

Reply:
...have been found...

Reply:
https://github.com/redhat-developer/yaml-language-server/pull/308